src/OrbitBase/include/OrbitBase/ToString.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_TO_STRING_H_
+#define ORBIT_BASE_TO_STRING_H_
+
+#include <algorithm>
+#include <string>
+
+// Wrapper around std::to_string which adds support for strings.
+
+namespace orbit_base {
```

Comment:
> This is mainly useful for day-to-day debugging, and we don't expect too
many occurences to make their way to the official codebase. That being
said, there are cases where we actually want to log information in the
exact "name = value" syntax. We will rely on code reviews to push back
on misuse.

This is the reason why we removed it in the first place. You are saying that it should be used rarely and that it's prone to misuse. Then we should not have it:
- "we don't expect too many occurrences": If it's to be used rarely, `ORBIT_LOG("name =  %whatyouneed", name)` is enough.
- "We will rely on code reviews to push back on misuse": This doesn't work. What constitutes misuse? How is someone going to know that it should be used rarely?

Also, I find it weird to have a printf-debugging macro in the codebase&mdash;everyone does printf-debugging differently. But @beckerhe thinks it differently so clearly this is just my opinion.

So this said, you want it back. Then let's keep it out of `OrbitBase`, as something in `OrbitBase` is encouraged to be used, not the opposite. A separate module sounds reasonable. But note: the namespace doesn't show in the macro name. So in addition I would prefer something like `ORBIT_DEBUG_VAR`.

Reply:
>I'm not sure if this should be in orbit_base. The wchar-string to char-string only works for ASCII characters and certain ASCII-compatible encoding.  I think it's okay to have something to make debugging easier but I'm worried that someone might use this function outside of the ORBIT_LOG_VAR macros.

It would actually be useful non-debug contexts. Dealing with Windows data structure, we often run into wide character ASCII strings which we sometimes need to convert to std::string. Getting thread names is one example. We use this pattern now: `std::string str(wstr.begin(), wstr.end())`, which is equivalent to what we have in ToString, but it generates a warning about the truncation. I think it's useful to have a dedicated conversion function, and I think orbit_base is a reasonable place to have it. We could actually warn at runtime if the truncation is losing information, if someone decided to name a thread using emojis for example. Wdyt?

Reply:
>This is the reason why we removed it in the first place. You are saying that it should be used rarely and that it's prone to misuse. Then we should not have it:

I disagree. Anything can be misused, ORBIT_LOG, lambdas, passing by value, protos, etc, etc. And we catch misuses daily in code reviews. Here we have an inoffensive macro that helps day-to-day debugging by minimizing friction when outputting internal state. It promotes validating assumptions as we are developing. I really think this should be encouraged. Having this in Logging.h allows universal access. A developer won't have to add dependency to a debug module, and usually won't even need to add an include, and that's by design, again to minimize friction.

Of course the exact same logging can be done using `ORBIT_LOG("name = %whatyouneed", name)`, which we would still have to push back against in a code review for misuse btw, but the only purpose of the (old) new macro is to not have to type more than we could and not have to think about "whatyouneed".

Reply:
> Having this in Logging.h allows universal access. A developer won't have to add dependency to a debug module, and usually won't even need to add an include, and that's by design, again to minimize friction.

Hmm, you kind of lost me here. It seems that we agree that for actual logging `ORBIT_LOG` is to be preferred, and you are saying that the printf-debugging macros should be in `Logging.h` because it's the most likely header to be already included or transitively included. Not sure I agree with the reasoning. I'll leave the hot potato to @beckerhe 's wisdom and knowledge on string encoding.

(It could still be `ORBIT_DEBUG_VAR` instead of `ORBIT_LOG_VAR` to show it's a debugging macro by the way, regardless of placement.)

Reply:
@beckerhe , @dpallotti , thanks for your insights and suggestions. I'll make some changes to the PR, maybe start a doc, and will get back to this ASAP. Thanks again.